runconfig/hostconfig.go

```diff
@@ -216,49 +248,55 @@ func (c *CapList) Slice() []string {
 	return c.caps
 }
 
+// NewCapList creates a CapList from a slice of string
 func NewCapList(caps []string) *CapList {
 	return &CapList{caps}
 }
 
+// HostConfig the non-portable Config structure of a container.
+// Here, "non-portable" means "dependent from the host we are running on".
+// Portable information *should* appear in Config.
 type HostConfig struct {
-	Binds            []string
-	ContainerIDFile  string
-	LxcConf          *LxcConfig
-	Memory           int64 // Memory limit (in bytes)
-	MemorySwap       int64 // Total memory usage (memory + swap); set `-1` to disable swap
-	CpuShares        int64 // CPU shares (relative weight vs. other containers)
-	CpuPeriod        int64
-	CpusetCpus       string // CpusetCpus 0-2, 0,1
-	CpusetMems       string // CpusetMems 0-2, 0,1
-	CpuQuota         int64
-	BlkioWeight      int64 // Block IO weight (relative weight vs. other containers)
-	OomKillDisable   bool  // Whether to disable OOM Killer or not
-	MemorySwappiness int64 // Tuning container memory swappiness behaviour
-	Privileged       bool
-	PortBindings     nat.PortMap
-	Links            []string
-	PublishAllPorts  bool
-	Dns              []string
-	DnsSearch        []string
-	ExtraHosts       []string
-	VolumesFrom      []string
-	Devices          []DeviceMapping
-	NetworkMode      NetworkMode
-	IpcMode          IpcMode
-	PidMode          PidMode
-	UTSMode          UTSMode
-	CapAdd           *CapList
-	CapDrop          *CapList
-	GroupAdd         []string
-	RestartPolicy    RestartPolicy
-	SecurityOpt      []string
-	ReadonlyRootfs   bool
-	Ulimits          []*ulimit.Ulimit
-	LogConfig        LogConfig
-	CgroupParent     string // Parent cgroup.
-	ConsoleSize      [2]int // Initial console size on Windows
-}
-
+	Binds            []string         // List of volume bindings for this container
+	ContainerIDFile  string           // File (path) where the containerId is written
+	LxcConf          *LxcConfig       // Additional lxc configuration
+	Memory           int64            // Memory limit (in bytes)
+	MemorySwap       int64            // Total memory usage (memory + swap); set `-1` to disable swap
+	CPUShares        int64            `json:"CpuShares"` // CPU shares (relative weight vs. other containers)
```

Comment:
Wasn't sure about adding the ``json:…`` notation here, but as it is in the API and consumed/produced already by other external components, I though it would be best to make it so the json in the API doesn't change (to not break external tools).